data_structures.rst

```diff
@@ -75,13 +69,7 @@ Actuators:
         the same code and machinery on different beamlines, where things are
         implemented in different ways.
 
-        # NB upper_limit and lower_limit are given separately to make it easier
-        #    to support the pair-of-floats type.
-
-        # NB allowed_values, if not empty, gives the allowed values.
```

Comment:
Ah yes, I was not entirely sure what you meant by that (so I wanted to discuss it):

 - Its quite clear that we need to supply a list inorder to use allowed values (if not empty)
 - For float however, you round down/up to closest value in the list ?

Reply:
@marcus-oscarsson 

I had in mind the case where you had a limited set of e.g. apertures, or transmission values at some beamlines, but others either had stepless setting, or preferred to let people type in arbitrary float values anyway. For that the semantics of a float enumeration ought to be to allow arbitrary values (within the allowed range?) but to round to the nearest allowed value when there was a set of allowed values to choose from. That would allow you to set e.g. a transmission to 87% and get the nearest allowed value, without having to match exactly one of the allowed steps.

Reply:
I see, that is indeed reasonable. I was not sure that is what you meant.